docs/shrc_diag_util.md

```diff
@@ -22,104 +22,124 @@
+If you are instrumenting bytecode by using a JVMTI agent or `java.lang.instrument` agent, the following rules apply:
 
-    - `1` : the ID of the VM that wrote this data.
-    - `0x2234F7DC` : the address where the metadata about the class is stored.
-    - `ROMCLASS` : the type of data that was stored.
-    - `java/lang/Runnable` : the name of the class.
-    - `0x213684A8` : the address where the class was stored.
-    - `Index 1` : the index in the class path where the class was loaded from.
-    - `0x2234FA6C` : the address of the class path against which this class is stored.
+- Redefined classes are never stored in the cache.
+- Retransformed classes are stored only if you specify the `-Xshareclasses:cacheRetransformed` suboption when you start your application.
 
-: Stale classes are marked with `!STALE!`. Any partition or modification context that is used when the class is stored is also shown.
+If a running application uses its own class loader and you are using a `SecurityManager`, you must grant the class loader permission to `SharedCachePermission` before they can share classes. For more information, see [Support for custom class loaders](shrc.md#support-for-custom-class-loaders).
 
-### AOT methods
+In very rare cases, problems with finding or storing classes might be due to cache corruption. If the VM detects that a cache is corrupt, it attempts to destroy the cache and re-create it. If the VM cannot re-create the cache it starts only if the `-Xshareclasses:nonfatal` suboption is specified on the command line. Try using the `-Xshareclasses:destroyAll` cache utility to remove the cache and re-create it.
 
-: This example shows an entry for one AOT-compiled method:
+## Generating verbose output
 
-        1: 0x540FBA6A AOT: loadConvert
-                for ROMClass java/util/Properties at 0x52345174
+A number of `-Xshareclasses` suboptions are available for generating verbose output during class data sharing operations, which can help you identify the root cause of a problem.
 
-    - `1` : the ID of the VM that wrote this data.
-    - `0x540FBA6A` : the address where the data is stored.
-    - `AOT` : the type of data that was stored.
-    - `loadConvert` : the method for which AOT-compiled code is stored.
-    - `java/util/Properties` : the class that contains the method.
-    - `0x52345174` : the address of the class that contains the method.
+### `verbose`
 
-: Stale methods are marked with `!STALE!`.
+The [`-Xshareclasses:verbose`](xshareclasses.md#verbose) suboption provides basic output on cache usage. In the following example, a persistent cache is opened and attached to the VM for class sharing. Information is provided about the size of the cache, any soft limits set, and the storage of AOT and JIT data.
 
-### URLs and tokens
+```
+java -Xshareclasses:name=myCache,verbose HelloWorld
+[-Xshareclasses persistent cache enabled]
+[-Xshareclasses verbose output enabled]
+JVMSHRC237I Opened shared classes persistent cache Cache1
```

Comment:
I forgot to change the cache name in the output text when providing this example. Should be "myCache" this line and following line.